### ISSUE TITLE
Update on Windows arm64 installed x64  upgrade

### ISSUE THREAD

**Issue by mirogl** [marticliment/WingetUI#448]
Hi,
i tried WingrtUI and the first try was to update my existing packages. 
But there seems to be a bug, since WingetUI is updateing everything to the x64 version, buy i have arm64 Versions installed.

For example - WingetUI installed the latest 7-zip x64 version - installed was 7-zip arm64 version.
If i execute manually via Command Line:

winget install 7zip.7zip

Then winget will install the correct arm64 version https://www.7-zip.org/a/7z2107-arm64.exe 

So there seems to be some bug in WingetUI, since winget command line works correctly.
Thanks
Miro

**Comment by martinet101** [comment]
hi @mirogl,

this issue is not with WingetUI itself, but rather with the Winget Command-line utility.
If you were to run `winget upgrade --all`, the same would happen.

I recommend you to post an issue on https://github.com/microsoft/winget-cli, since this issue can't be fixed from WingetUI.

I'm sorry for not being able to help :(

**Comment by mirogl** [comment]
Hi,
no - i created this bug, because winget works (v1.3.2691). I executed "winget upgrade --all" before opening this ticket and it installed automatically the arm64 versions again  - for example 7-Zip & Teams. With WingetUI it installed the x64 versions of 7-zip & Teams.
Therefore i was able to fix the wrong installed x64 packages by executing "winget upgrade --all"

Thanks
Miro

**Comment by martinet101** [comment]
do you know if the winget installation you have in your system is arm64 or emulated x64?

**Comment by mirogl** [comment]
Hi,
yes it is arm64.

winget --info
Windows-Paket-Manager v1.3.2691
Copyright (c) Microsoft Corporation. Alle Rechte vorbehalten.

Windows: Windows.Desktop v10.0.22621.819
Systemarchitektur: Arm64
Paket: Microsoft.DesktopAppInstaller v1.18.2691.0

Protokolle: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir

Links
-----------------------------------------------------------------------------------------
Datenschutzerklärung              https://aka.ms/winget-privacy
Lizenzvereinbarung                https://aka.ms/winget-license
Hinweise von Drittanbietern       https://aka.ms/winget-3rdPartyNotice
Startseite                        https://aka.ms/winget
Windows Store Nutzungsbedingungen https://www.microsoft.com/en-us/storedocs/terms-of-sale

Thanks
Miro

**Comment by martinet101** [comment]
Oh, I see where the issue.

The winget bundled version is a x64 version, and it does only install x64 software. 

I have opened an issue on Winget-cli: https://github.com/microsoft/winget-cli/issues/2731

**Comment by mirogl** [comment]
ok, thanks - i have updated the Winget-cli issue with the requested info.

**Comment by mirogl** [comment]
hi, 
I can test this for you - i have 3 arm64 Windows machines - but all are windows 11 ;-)
Couldn't you just offer the option not to installwinget with WingetUI and use the already systeminstalled version ?
I'm aksing, because your bundled version is older v1.3.0 vs v1.3.2691 - and i think many people just want a GUI for winget - but not two versions installed - maintain two versions.
Or give the option to switch between the bundled or systemwide installed version.

Thanks
Miro

**Comment by martinet101** [comment]
The issue with installed winget versions are languages.
The version bundled is a version from where languages other than english have been removed, so the output of the commands can be parsed properly. The issue is that winget is translated, but there's no command-line argument to show the output in english. 
Also, like this I can implement features thet depend on the latest winget version (soon 'im going to introduce labels, as soon as winget 1.4 is released), and I can make sure that WingetUI does not get broken due to some people not having updated wingetui.


If you want, there's a hidden flag that you can enable to force WingetUI to use system's winget installation.
 However, you'll need to have winget in english or winget packages will become broken.

To enable this flag you need to create an empty file called `UseSystemWinget` on the following durectory: `C:\Users\<yourusername>\.wingetui`

**Comment by mirogl** [comment]
Hi - thanks - i fully understand now, that it makes no sense to use a local version, where the output parsing can´t be parsed on different language.

**Comment by funkemunky** [comment]
> I mean, if they can't fix that i'll bundle arm winget releases as well, but it'd be better if winget already sorted that out, since I have no arm machine at home and I couldnn't test that

I'm happy to fork and test arm compiles for you since I have a Windows on Arm VM on my Macbook Pro.

**Comment by mirogl** [comment]
> Hello, a new commit has been pushed, introducing arm winget builds. Can you please test the latest code enabling the "Use ARM winget" option under experimental settings?

Hi, should we wait for the newest release, which will include this commit or do you expect us to build it on our own from the source code ?

**Comment by marticliment** [comment]
You can just run the source code. 

If you are used to python, is just installing the requirements.txt and double-clicking the __init__.py file.

Then Enable the arm setting under experimental features, quit wingetui, launch it again and atempt to install some arm package to see what happens

**Comment by funkemunky** [comment]
> You can just run the source code.
> 
> If you are used to python, is just installing the requirements.txt and double-clicking the **init**.py file.
> 
> Then Enable the arm setting under experimental features, quit wingetui, launch it again and atempt to install some arm package to see what happens

On it

**Comment by marticliment** [comment]
There is now a beta release with the arm winget binaries

**Comment by mirogl** [comment]
> There is now a beta release with the arm winget binaries

Thanks - tried to run the new Beta - installed it with non Admin User and got

<img width="304" alt="image" src="https://github.com/marticliment/WingetUI/assets/46027543/60aede3c-4d2a-44aa-99d4-f74d2ad97a96">

**Comment by marticliment** [comment]
Try downloading and installing the beta again (i have updated the installer)

**Comment by mirogl** [comment]
> Try downloading and installing the beta again (i have updated the installer)

Hi,
Thanks . now the Gui started - i choosed the "Use ARM winget" option under experimental settings" - but it still installed Spotify x86_x64 for my installed Spotify arm64.

And yes - i restarted WinGet after setting this option and is was still active after the Restart.

![image](https://github.com/marticliment/WingetUI/assets/46027543/faab31ba-bbe7-427b-be14-009df902fcab)

![image](https://github.com/marticliment/WingetUI/assets/46027543/45b74673-d720-4986-845e-1fae4408b6f6)

![image](https://github.com/marticliment/WingetUI/assets/46027543/d4cf9812-80d4-4755-af7b-a9f33de45472)

Thanks
Miro

**Comment by marticliment** [comment]
> > Try downloading and installing the beta again (i have updated the installer)
> 
> Hi, Thanks . now the Gui started - i choosed the "Use ARM winget" option under experimental settings" - but it still installed Spotify x86_x64 for my installed Spotify arm64.
> 
> And yes - i restarted WinGet after setting this option and is was still active after the Restart.
> 
> ![image](https://user-images.githubusercontent.com/46027543/253368726-faab31ba-bbe7-427b-be14-009df902fcab.png)
> 
> ![image](https://user-images.githubusercontent.com/46027543/253368927-45b74673-d720-4986-845e-1fae4408b6f6.png)
> 
> ![image](https://user-images.githubusercontent.com/46027543/253369133-d4cf9812-80d4-4755-af7b-a9f33de45472.png)
> 
> Thanks Miro

It might be a dumb question, but was spotify installed as arm, right?
Can you please check that effectively, when WingetUI calls winget, the arm64 version is being called?

**Comment by mirogl** [comment]
Hi,
yes - the Arm64 Version of Spotify was installed before the update. You can see in the screenshot from my last post, which Package has been downloaded - it was spotify_installer from the x86_x64 path. 
Thanks

**Comment by marticliment** [comment]
> Can you please check that effectively, when WingetUI calls winget, the arm64 version is being called

Sorry, I meant the winget.exe process. When installing, the wiinget.exe process should be there

**Comment by mirogl** [comment]
Hi,
I checked this - when i try to uninstall a package, the x64 winget is used.
![image](https://github.com/marticliment/WingetUI/assets/46027543/55914d72-a910-45ae-9ec8-e710ed1df24c)

**Comment by marticliment** [comment]
Can you please share your log?

**Comment by mirogl** [comment]
Hi, I tried both now - install Spotifay via WingetUI (after unsinstalling Spotify) and Uninstall Windows Family - now it works correctly. The Only thing I mafe was a reboot. Perhapps the problem was, that I just upgraded WingetUI Beta and didn't made a Reboot - ist this necessary after the Upgrade, because any system environment variables are updatet ?

Now from the Log everything is ok - also from the Task Manager - only the arm64 Bit winget is used, which I can see from the protocoll. So a reboot fixed the problem.
